### PR TITLE
Using Jinja2 templates for generating ssg.py content

### PR DESCRIPTION
**Pull request by cptanalatriste**
I managed to get Jinja2 working for generating the ssg.py content. That way, all the "frontend" code is contained in the Jinja template, and in the "backend" (`make.py`) we just provide what the template needs.

This is a rough initial prototype. Before doing a clean-up, I wanted to get your feedback :)

### CODE REVIEW

File: sqlsynthgen/make.py [hunk 1]
```diff
@@ -59,38 +52,46 @@ def _orm_class_from_table_name(tables_module: Any, full_name: str) -> Optional[A
...
+
+        column_info.append(column_data)
+    return column_info, columns_covered
 
 
-def _add_default_generator(content: str, tables_module: ModuleType, column: Any) -> str:
+def _add_default_generator(tables_module: ModuleType, column: Any) -> Dict[str, Any]:
     """Append a default generator to content, for the given column."""
-    content += INDENTATION * 2
+    column_data: Dict[str, Any] = {"primary_key": False}
```
Comment (mhauru) [inline sqlsynthgen/make.py:90]: This and the next few lines could just be `"primary_key": column.primary_key`.

**Comment by mhauru** [comment]
I took a quick look to get an idea of the approach, and I really like it. I'm happy to go ahead with it. I also like Iain's suggestion in #14 of running the output of Jinja through Black.

File: sqlsynthgen/make.py (hunk 1 above)
Comment (cptanalatriste) [reply sqlsynthgen/make.py:90]: Makes sense!

File: sqlsynthgen/templates/ssg.py.template
Comment (Iain-S) [inline sqlsynthgen/templates/ssg.py.template:1]: Not sure whether calling it `ssg.j2` or `ssg.jinja2` might get some better syntax highlighting?

Comment (cptanalatriste) [reply sqlsynthgen/templates/ssg.py.template:1]: `ssg.py.j2` looks really good with the [Better Jinja plugin](https://marketplace.visualstudio.com/items?itemName=samuelcolvin.jinjahtml).

**Comment by cptanalatriste** [comment]
Here's a new version of the PR. This one is looking better:
- Using black to format the output of Jinja2.
- Rename the template so syntax highlighting works in VS Code.
- Simplified the template, using dataclasses to represent generators and their attributes.

**Review by Iain-S — APPROVED** [review]

**Comment by mhauru** [comment]
I did the merge of main into this branch. @cptanalatriste please check that you're happy with the result and make e.g. any style adjustments you like to the stuff I added while merging (for instance a new section in the template). I did not attend to the points raised above by @Iain-S. We did the review together, so I have nothing to add to his comments + approval.

**Pull request merged**